primitives/consensus/babe/src/lib.rs

```diff
@@ -94,6 +96,23 @@ pub fn make_transcript(
 	transcript
 }
 
+/// Make a VRF transcript data container
+#[cfg(feature = "std")]
+pub fn make_transcript_data(
```

Comment:
Is it possible to test that `make_transcript_data().make_transcript()` == `make_transcript()`?

Doesn't seem that `merlin::Transcript` has a `PartialEq` implementation, but as a hack you could check that the built RNG produces the same bytes: https://docs.rs/merlin/2.0.0/merlin/struct.TranscriptRngBuilder.html

`OsRng` will not be deterministic but `rand_chacha` provides a seedable, deterministic, `CryptoRng` implementation that can be used in the `dev-dependencies` here.

Reply:
cc @burdges Is this a reasonable way of checking transcript equality?

Reply:
These merlin transcripts are hash functions that do not stop running when you extract some data, so yes you can test equality by taking enough data out.  You can `clone` and then call [`challenge_bytes`](https://docs.rs/merlin/2.0.0/merlin/struct.Transcript.html#method.challenge_bytes) or 
```
    let test = |t| -> [u8; 16] {
        let mut b = [0u8; 16];
        t.build_rng()
        .finalize( &mut ChaChaRng::from_seed([0u8;32]) )
        .fill_bytes(&mut b);
        b
    }
    test(t1) == test(t2)
```

Reply:
I implemented this test in the vrf module in sp_core.